protocol/src/main/java/org/geysermc/mcprotocollib/protocol/codec/MinecraftCodecHelper.java

```diff
@@ -203,13 +203,18 @@ public CompoundTag readAnyTagOrThrow(ByteBuf buf) throws IOException {
     }
 
     @Nullable
-    public <T extends Tag> T readAnyTag(ByteBuf buf, Class<T> expected) throws IOException {
-        Tag tag = NBTIO.readAnyTag(new InputStream() {
-            @Override
-            public int read() {
-                return buf.readUnsignedByte();
-            }
-        });
+    public <T extends Tag> T readAnyTag(ByteBuf buf, Class<T> expected) {
+        Tag tag;
+        try {
+            tag = NBTIO.readAnyTag(new InputStream() {
+                @Override
+                public int read() {
+                    return buf.readUnsignedByte();
+                }
+            });
+        } catch (IOException e) {
+            throw new NBTRuntimeException(e);
```

Comment:
this new exception class doesn't provide any more information - i'd just use an illegal argument exception